### ISSUE TITLE
WCAG: Remove 'Tiled Gallery' image title attribute if the alt attribute is empty

### ISSUE THREAD

**Issue by davetgreen** [Automattic/jetpack#2679]
As per http://www.w3.org/TR/WCAG20-TECHS/H67.html the title attribute should be absent if the image has no alt attribute to ensure that screen readers can ignore the image(s).

Issue raised when using Total Validator Pro during local development. My PR successfully resolves the issue and ensures the markup passes validation.

**Comment by dereksmart** [comment]
Looks good -- thanks for the patch @davetgreen

**Comment by mrwweb** [comment]
I'd recommend completely removing the title attributes. WordPress core has been doing this over the past few versions and I expect it to eventually not output any title attributes at all. I think Jetpack should do the same. They rarely improve accessibility and are definitely not-accessible to many users (most screen readers, all touch devices, etc.).

**Comment by davetgreen** [comment]
@mrwweb there's some discussion over in #2680 about this at the moment, but I agree with they have little benefit.

**Comment by jeherve** [comment]
Reopening this issue, as per @odie2's comment on the PR (#2873):
https://github.com/Automattic/jetpack/pull/2873#issuecomment-154722521

----

Latest commit **have to** be _immediately_ reverted!

**First reason**:
Most of images are `P1040480.jpg` and `DSC06635.jpg`, so what sense makes to set `alt="DSC06635.jpg"` to make noise for screen readers?
In most of galleries (like my - even up to 200 random photos of past event) if Editor doesn't described photos, it's better to leave empty `alt` tag (as was before) as system can't provide any alternative info if human didn't provided it.
> Some images, such as invisible pixels or images used in toolbars, are used solely for layout purposes and provide no content. The ALT tag for these images should be blank ( <alt= "" >) so that screen readers will ignore them altogether.
> http://accessibility.psu.edu/images/imageshtml/ (header 2: "Alt Text for Filler or Decorative Images")

**Second reason**:
> If a text description is already provided for an image within the main text of the page, then the ALT tag can just provide a summary of the image, not a full description.
> http://accessibility.psu.edu/images/imageshtml/ (header 2: "Summary Alt Text")

Currently having a `title` set and no `alt` set gives cloned text:
`<div class="tiled-gallery-item"><img title="My title" alt="My title"/></div>`

This commit will mess in whoooole search engines indexes of tiled galleries, will annoying screen reader's users... I am luckily, that I readed changelog before updating as I am participate in WCAG competition for my country...

----

It seems we have 2 issues here:

**1. Alt attributes**

I think a good alternative would be to use an approach similar to Core for Alt titles, and fallback to the caption when an image doesn't include any Alt value. If no caption exists either, we fallback to the title.
https://core.trac.wordpress.org/browser/tags/4.3.1/src/wp-includes/media.php#L755

We've in fact done something somehow similar in the past in Jetpack, in #2631.

Checking for a caption would ensure that the alt attribute is a bit more useful.

**2. Title attributes**

If we follow Core's approach, we can remove them completely in galleries, much like @mrwweb  suggested:
https://core.trac.wordpress.org/browser/tags/4.3.1/src/wp-includes/media.php#L1117

Let's look at how a gallery image including all parameters looks like in Core right now:

![screen shot 2015-11-09 at 1 50 52 pm](https://cloud.githubusercontent.com/assets/426388/11034084/ec64360e-86e8-11e5-9a81-51a956f43144.png)

![screen shot 2015-11-09 at 1 47 50 pm](https://cloud.githubusercontent.com/assets/426388/11034068/cefc2928-86e8-11e5-90af-76c5b8b5145d.png)

*-> No title attribute, an Alt attribute.*

If I remove the Alt attribute from that image:

![screen shot 2015-11-09 at 1 51 23 pm](https://cloud.githubusercontent.com/assets/426388/11034088/fed703c0-86e8-11e5-8b52-f7b110123f53.png)

![screen shot 2015-11-09 at 1 52 08 pm](https://cloud.githubusercontent.com/assets/426388/11034101/1a219bcc-86e9-11e5-8d66-48b838dd3f46.png)

*-> No title attribute, the alt attribute now uses the Caption value.*

If I remove the Caption value:

![screen shot 2015-11-09 at 1 53 01 pm](https://cloud.githubusercontent.com/assets/426388/11034114/3949e392-86e9-11e5-806c-53a1d7fe0a2b.png)

![screen shot 2015-11-09 at 1 53 59 pm](https://cloud.githubusercontent.com/assets/426388/11034128/5a53c17a-86e9-11e5-92c3-7c37f2cd741c.png)

*-> No title attribute, the alt attribute now uses the title value.*

If I remove the title value:

![screen shot 2015-11-09 at 1 55 41 pm](https://cloud.githubusercontent.com/assets/426388/11034151/96c924a6-86e9-11e5-9c5f-9192f492ffc2.png)

![screen shot 2015-11-09 at 1 57 04 pm](https://cloud.githubusercontent.com/assets/426388/11034170/c9b3b836-86e9-11e5-9b12-7fecd5af169b.png)

*-> No title attribute, empty alt attribute.*

So, to conclude:

**Let's do it like Core** :)

1. Alt attribute: get it from Alt value > caption > title > No alt value.
2. Title attribute: none.

**Comment by odie2** [comment]
@jeherve Yea, it should be exactly as you wrote about caption and title.

I missed a little with my __first reason__ that it's rather Core issue, so I should filter it.
However, they did everything to make me think that isn't issue, just analyze:

![image](https://cloud.githubusercontent.com/assets/9401878/11036614/ec355b02-86f9-11e5-84ad-0c66ed2b4c23.png)
> Add Medium (in visual editor) modal view

![image](https://cloud.githubusercontent.com/assets/9401878/11036690/5acd6550-86fa-11e5-8df2-61c5b695518a.png)
> Edit Medium (in visual editor) modal view

So I should report it as wrong labelled + accessibility issue in Core. I just followed that Tiled Galleries before that commit were accessibility right.

**Comment by odie2** [comment]
You can follow and discuss on this topic here: https://core.trac.wordpress.org/ticket/34635
